pkg/virt-launcher/virtwrap/cmd-server/server.go

```diff
@@ -289,7 +290,11 @@ func (l *Launcher) GetDomain(_ context.Context, _ *cmdv1.EmptyRequest) (*cmdv1.D
 	}
 
 	if len(list) > 0 {
-		if domain, err := json.Marshal(list[0]); err != nil {
+		decoratedDomain := list[0]
+		decoratedDomain.Status.OSInfo = l.domainManager.GetGuestOSInfo()
+		decoratedDomain.Status.Interfaces = getInterfacesStatus(l.domainManager, decoratedDomain.Spec.Devices.Interfaces)
```

Comment:
This seems equivalent to: `... = l.domainManager.InterfacesStatus(decoratedDomain.Spec.Devices.Interfaces)`

And it should remove the need to import the `api`.

BTW, you can rename `decoratedDomain` to something shorter like `d` or `domainObj`.